scripts/Muon/GUI/Common/corrections_tab_widget/corrections_presenter.py

```diff
@@ -0,0 +1,189 @@
+                self.view.warning_popup(error)
+
+    def _handle_selected_table_is_invalid(self) -> None:
+        """Handles when the selected dead time table workspace is invalid."""
+        # Triggers handle_dead_time_from_selector_changed
+        self.view.set_dead_time_from_data_file_selected()
+
+    def handle_dead_time_browse_clicked(self) -> None:
+        """User selects the option to Browse for a nexus file to load dead times from."""
+        filename = self.view.show_file_browser_and_return_selection(["nxs"], [""], multiple_files=False)[0]
+        if filename != "":
+            name = self._load_file_containing_dead_time(filename)
+            if name is not None:
+                self.view.populate_dead_time_workspace_selector(get_table_workspace_names_from_ADS())
+                error = self.model.validate_selected_dead_time_workspace(name)
+                if error == "":
+                    self.view.switch_to_using_a_dead_time_table_workspace(name)
+                else:
+                    self.view.warning_popup(error)
+
+    def handle_corrections_complete(self) -> None:
+        """When the corrections have been calculated, update the displayed dead time averages."""
+        self.update_dead_time_info_text_in_view()
+
+    def update_dead_time_info_text_in_view(self) -> None:
+        """Update the dead time info label in the view."""
+        if self.model.is_dead_time_source_from_data_file() or self.model.is_dead_time_source_from_workspace():
+            self.view.set_dead_time_average_and_range(self.model.dead_times_range(), self.model.dead_times_average())
+        else:
+            self.view.set_dead_time_info_text("No dead time correction")
+
+    def set_dead_time_source_to_from_file(self) -> None:
+        """Sets the dead time source to be from the data file and notifies the GUI to recalculate the corrections."""
+        self.model.set_dead_time_source_to_from_file()
+        self._notify_perform_dead_time_corrections()
+
+    def set_dead_time_source_to_from_ads(self) -> None:
+        """Sets the dead time source to be the ADS and notifies the GUI to recalculate the corrections."""
+        self.model.set_dead_time_source_to_from_ads(self.view.selected_dead_time_workspace())
+        self._notify_perform_dead_time_corrections()
+
+    def set_dead_time_source_to_none(self) -> None:
+        """Sets the dead time source to be none and notifies the GUI to recalculate the corrections."""
+        self.model.set_dead_time_source_to_none()
+        self._notify_perform_dead_time_corrections()
+
+    def _load_file_containing_dead_time(self, filename: str) -> str:
+        """Attempts to load a Nexus cycle file containing a dead time table workspace."""
+        try:
+            name = load_dead_time_from_filename(filename)
+        except Exception:
+            self.view.warning_popup("The file provided has an unexpected format. A Nexus cycle file must be provided.")
```

Comment:
whats a "Nexus cycle file"?

Reply:
I am trying to say the nexus file you get from the cycle folder of the data archive. Is there a better phrasing for that?

Reply:
In theory, I think it should be possible to load a deadtime table from a different cycle. I think it just needs to be an appropriate nexus file. I think that just means one with the same number of detectors and a similar/same set up

Reply:
Ok, I have created an issue to do this separately as this is not possible in the current version of Mantid #31891

Reply:
from a practical stand point, I dont see why anyone would use things from different cycles. 

Maybe we should just change the error to:

The file provided is not of the expected format. The file should be of the same instrument and cycle as the raw data.